app/src/main/java/com/example/tasky/agenda/data/DateGeneratorImpl.kt

```diff
@@ -0,0 +1,43 @@
+package com.example.tasky.agenda.data
+
+import com.example.tasky.agenda.domain.DateGenerator
+import com.example.tasky.agenda.presentation.model.Day
+import java.util.Calendar
+import java.util.Calendar.DAY_OF_MONTH
+import java.util.Locale
+import javax.inject.Inject
+
+class DateGeneratorImpl @Inject constructor() : DateGenerator {
+
+    override fun getWeek(): List<Day> {
```

Comment:
This method is weird
It will return the week of the current day

But you need to also be able to change the date, since you can modify that on the Agenda screen

Also, What you can do to avoid having presentation stuff here, is something like:


Return a list of "LocalDate" based on the LocalDate sent in the parameters

And then on the presentation layer, have a "mapper" that maps each LocalDate, to a Day object

So this method's signature would be something like:

```
fun getWeekFromDate(date: LocalDate): List<LocalDate>
```